src/VisualizationTests/HelixRenderPackageTests.cs

```diff
@@ -157,6 +158,398 @@ public void HelixRenderPackage_MeshColorBufferCheck()
+
+            //Update colors
+            p.UpdateMeshVertexColorForRange(0,3,255, 255, 255, 255);
+
+            Assert.AreEqual(p.MeshVertexColors, testColors);
+        }
+
+        [Test]
+        public void HelixRenderPackage_UpdateMeshVertexColorForRange_ThrowsForBadRange()
+        {
+            var p = new HelixRenderPackage();
+
+            PushQuadIntoPackage(p);
+
+            //Try to update without any colors set
+            try
+            {
+                p.UpdateMeshVertexColorForRange(0, 3, 255, 255, 255, 255);
+            }
+            catch (Exception e)
+            {
+                Assert.AreEqual(e.GetType(), typeof(Exception));
+            }
+
+            Assert.AreEqual(0, p.MeshVertexColorCount);
+
+            var testColors = WhiteByteArrayOfLength(6);
+            //Set colors
+            p.AppendMeshVertexColorRange(testColors);
+
+            //Try to update colors with wrong range
+            try
+            {
+                p.UpdateMeshVertexColorForRange(0, 7, 255, 255, 255, 255);
+            }
+            catch (Exception e)
+            {
+                Assert.AreEqual(e.GetType(), typeof(Exception));
+            }
+
+            Assert.AreEqual(6, p.MeshVertexColorCount);
+
+            //Try to update colors with index error
+            try
+            {
+                p.UpdateMeshVertexColorForRange(6, 0, 255, 255, 255, 255);
+            }
+            catch (Exception e)
+            {
+                Assert.AreEqual(e.GetType(), typeof(Exception));
+            }
+
+            Assert.AreEqual(6, p.MeshVertexColorCount);
+        }
+
+        [Test]
+        public void HelixRenderPackage_AppendLineVertexColorRange_Success()
+        {
+            var p = new HelixRenderPackage();
+
+            PushLineIntoPackage(p);
+
+            var testColors = WhiteByteArrayOfLength(2);
+
+            p.AppendLineVertexColorRange(testColors);
+
+            Assert.AreEqual(p.LineStripVertexColors, testColors);
+
+            PushLineIntoPackage(p);
+
+            p.AppendLineVertexColorRange(testColors);
+
+            var totalColors = WhiteByteArrayOfLength(4);
+
+            Assert.AreEqual(p.LineStripVertexColors, totalColors);
+        }
+
+        [Test]
+        public void HelixRenderPackage_AppendLineVertexColorRange_ThrowsForBadRange()
+        {
+            var p = new HelixRenderPackage();
+
+            PushLineIntoPackage(p);
+
+            var testColors = WhiteByteArrayOfLength(1);
+
+            try
+            {
+                p.AppendLineVertexColorRange(testColors);
+            }
+            catch (Exception e)
+            {
+                Assert.AreEqual(e.GetType(), typeof(Exception));
+            }
+
+            Assert.AreEqual(0, p.LineVertexColorCount);
+
+            testColors = WhiteByteArrayOfLength(3);
+
+            try
+            {
+                p.AppendLineVertexColorRange(testColors);
+            }
+            catch (Exception e)
```

Comment:
Can we make these exceptions thrown be more specific?

Reply:
I think I can cover these in the next PR as I add more test coverage.